Viper Practice/src/networking/API.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  API.swift
+//  NetworkLayer
+//
+//  Created by Malek TRABELSI on 08/01/2023.
+//
+
+import Foundation
+
+class API {
+    private let apiConfig: APIConfig
+    private let apiFetcher: HTTPClient
+
+    init(apiConfig: APIConfig, apiFetcher: HTTPClient) {
+        self.apiConfig = apiConfig
+        self.apiFetcher = apiFetcher
+    }
+}
+
+extension API {
```

Comment:
there should be a separate class for FruitesApi

Reply:
create separate class FruitesApi and make API class generic